packages/core/test/test-util/test-services/v4/test-service/function-imports.ts

```diff
@@ -87,6 +89,46 @@ export function testFunctionImportEntityReturnTypeCollection(parameters: TestFun
   return new FunctionImportRequestBuilderV4('/sap/opu/odata/sap/API_TEST_SRV', 'TestFunctionImportEntityReturnTypeCollection', (data) => transformReturnValueForEntityListV4(data, TestEntity), params);
 }
 
+/**
+ * Type of the parameters to be passed to [[testFunctionImportSharedEntityReturnType]].
+ */
+export interface TestFunctionImportSharedEntityReturnTypeParameters {
+}
+
+/**
+ * Test Function Import Shared Entity Return Type.
+ *
+ * @param parameters - Object containing all parameters for the function import.
+ * @returns A request builder that allows to overwrite some of the values and execute the resultng request.
+ */
+export function testFunctionImportSharedEntityReturnType(parameters: TestFunctionImportSharedEntityReturnTypeParameters): FunctionImportRequestBuilderV4<TestFunctionImportSharedEntityReturnTypeParameters, any> {
+  const params = {
+
+  }
+
+  return new FunctionImportRequestBuilderV4('/sap/opu/odata/sap/API_TEST_SRV', 'TestFunctionImportSharedEntityReturnType', (data) => throwErrorWhenReturnTypeIsUnionType('TestFunctionImportSharedEntityReturnType'), params);
+}
```

Comment:
The function import adjusted manually, where the return type is set to `any` and the deserialise function is replaced by a function that throws an error: `throwErrorWhenReturnTypeIsUnionType`;

Reply:
@marikaner , as discussed, the error is thrown via the client code.